### ISSUE TITLE
HeadObjectCommand throws TypeError: Cannot read properties of null (reading 'getReader')

### ISSUE THREAD

**Issue by max-lt** [aws/aws-sdk-js-v3#4398]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

With HeadObjectCommand (and S3Client) 

I got the followinf error: `TypeError: Cannot read properties of null (reading 'getReader')`
Following the stacktrace it's caused by the output body being null

Stacktrace: 
```
collectStream
Object.streamCollector
collectBody
deserializeAws_restXmlHeadObjectCommand
deserialize
```

```javascript
const deserializeAws_restXmlHeadObjectCommand = async (output, context) => {
    if (output.statusCode !== 200 && output.statusCode >= 300) {
        return deserializeAws_restXmlHeadObjectCommandError(output, context);
    }

   // ...

    await collectBody(output.body, context); // <- output.body is null
    return contents;
}
```

```typescript
import { GetObjectCommand, HeadObjectCommand, S3Client } from '@aws-sdk/client-s3';

//...
const s3 = new S3Client(config.s3);

function getObject(key: string) {
  return s3.send(new GetObjectCommand({ Bucket: config.bucket, Key: key }));
}

function headObject(key: string) {
   return s3.send(new HeadObjectCommand({ Bucket: config.bucket, Key: key }))
}
```

### SDK version number

@aws-sdk/package-name@3.262.0

### Which JavaScript Runtime is this issue in?

Browser

### Details of the browser/Node.js/ReactNative version

v18.11.0

### Reproduction Steps

Following the stacktrace it's caused by the output body being null

https://github.com/aws/aws-sdk-js-v3/blob/ceb0ef9f78a2b885c09ec0404d6435d048686fb3/clients/client-s3/src/protocols/Aws_restXml.ts#L5806

```javascript
const deserializeAws_restXmlHeadObjectCommand = async (output, context) => {
    if (output.statusCode !== 200 && output.statusCode >= 300) {
        return deserializeAws_restXmlHeadObjectCommandError(output, context);
    }

   // ...

    await collectBody(output.body, context); // <- output.body is null
    return contents;
}
```

### Observed Behavior

TypeError: Cannot read properties of null (reading 'getReader')
Stacktrace: 
```
collectStream
Object.streamCollector
collectBody
deserializeAws_restXmlHeadObjectCommand
deserialize
```

### Expected Behavior

No error

### Possible Solution

Check if body is null for this command

### Additional Information/Context

The project is compiled with webpack and ts-loader

**Comment by yenfryherrerafeliz** [comment]
Hi @max-lt, thanks for opening this issue. I am not sure which scenario causes the body to be null. Would you please be able to provide reproduction steps that describes better how to get this error? Could you please also let me know if you are using CloudFlare?. 

Reproduction steps:
- Create a new project ``` npm init -y```
- Install the dependencies:
    - ``` npm i @aws-sdk/client-s3 ```
- Create a new file: ``` touch src/index.js ```
- Copy and paste the following code:
```js
import {
    S3Client,
    HeadObjectCommand
} from "@aws-sdk/client-s3";

const client = new S3Client({
    region: process.env.TEST_REGION
});
const response = await client.send(new HeadObjectCommand({
    Bucket: process.env.TEST_BUCKET,
    Key: process.env.TEST_KEY
}));
console.log(response)

```
- Run the code: ``` node src/index.js ```

And, when the file does not exists I get an error indicating so, and when it does exist I get a successful response.

Thanks!

**Comment by max-lt** [comment]
Hello,
I'm not using Cloudflare workers but a similar runtime that I'm developping (compiled with `esbuild`) but it is possible to reproduce with Cloudflare workers, the S3 provider is Wasabi.

I adapted your code to make it work with Wasabi 

```javascript
import { S3Client, HeadObjectCommand } from "@aws-sdk/client-s3";

const config = {
  REGION: "eu-west-2",
  BUCKET: "aws-debug",
  SECRET_KEY: "...",
  ACCESS_KEY: "...",
};

const client = new S3Client({
  endpoint: "http://s3.eu-west-2.wasabisys.com",
  region: config.REGION,
  credentials: {
    accessKeyId: config.ACCESS_KEY,
    secretAccessKey: config.SECRET_KEY,
  },
});

const response = client
  .send(
    new HeadObjectCommand({
      Bucket: config.BUCKET,
      Key: "test.txt",
    })
  )
  .then(console.log);

```
The following code gives a 200 with node.

With Cloudflare workers:
  - `npm install wrangler`
  - `npx wrangler dev src/index.js`
  - Turn on local mode by pressing "L" :
  

```
 Unhandled Promise Rejection: TypeError: Cannot read properties of null (reading 'getReader')
    at collectStream (/Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/fetch-http-handler/dist-es/stream-collector.js:15:27)
    at Object.streamCollector (/Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/fetch-http-handler/dist-es/stream-collector.js:6:12)
    at collectBody (/Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/client-s3/dist-es/protocols/Aws_restXml.js:9863:20)
    at deserializeAws_restXmlHeadObjectCommand (/Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/client-s3/dist-es/protocols/Aws_restXml.js:4578:11)
    at deserialize (/Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/client-s3/dist-es/commands/HeadObjectCommand.js:48:16)
    at /Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/middleware-serde/dist-es/deserializerMiddleware.js:4:30
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at /Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/middleware-signing/dist-es/middleware.js:10:20
    at /Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/middleware-retry/dist-es/retryMiddleware.js:24:46
    at /Users/maxime/Documents/aws-debug/node_modules/@aws-sdk/middleware-logger/dist-es/loggerMiddleware.js:2:22
```

**Comment by AG-Labs** [comment]
I have been seeing the same. Inconsistent behaviour across different browsers. Safer was fine but some versions of chrome were seeing issues. Everything in the returned response seemed as expected just the calls on body that failed as it was null.

**Comment by severi** [comment]
This started failing for us with the newest chrome version (Version 113.0.5672.63 (Official Build) (arm64))

**Comment by jroru** [comment]
Also seeing this failure assosciated with the Chrome 113 release. In Chrome 113, [this response body](https://github.com/aws/aws-sdk-js-v3/blob/main/packages/fetch-http-handler/src/fetch-http-handler.ts#L103) is null.

As a workaround, we were able to pass a patched `requestHandler` to our SDK client constructor.

```ts
const sdkClient = new SdkClientConstructor({
  ...
  requestHandler: new PatchedFetchHttpHandler()
})
```

Where `PatchedFetchHttpHandler` is a copy+paste of [`FetchHttpHandler`](https://github.com/aws/aws-sdk-js-v3/blob/main/packages/fetch-http-handler/src/fetch-http-handler.ts#L103) plus this diff
```diff
- body: response.body,
+ body: response.body || new Uint8Array(),
```

**Comment by kuhe** [comment]
A fix was released in [v3.329.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.329.0). Use that version of the S3 Client, or otherwise ensure that your `@aws-sdk/fetch-http-handler` package is at that version (e.g. lockfile) to use this fix.

**Comment by Teddy939** [comment]
I was encountered same issue for Chrome 113.0.5672.63, but it disappeared for Chrome 113.0.5672.92 that is latest version of Chrome at 2023-05-09 (JST).

**Comment by trivikr** [comment]
> A fix was released in [v3.329.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.329.0). Use that version of the S3 Client, or otherwise ensure that your `@aws-sdk/fetch-http-handler` package is at that version (e.g. lockfile) to use this fix.

Closing as the fix was released in https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.329.0
